src/test/ui/did_you_mean/issue-46718-expected-dotdot-found-dotdotdot.stderr

```diff
@@ -0,0 +1,14 @@
+error: expected field pattern, found `...`
+  --> $DIR/issue-46718-expected-dotdot-found-dotdotdot.rs:21:55
+   |
+21 |             PersonalityInventory { expressivity: exp, ... } => exp
+   |                                                       ^^^ help: to omit remaining fields, use one fewer `.`: `..`
+
+error[E0027]: pattern does not mention field `instrumentality`
+  --> $DIR/issue-46718-expected-dotdot-found-dotdotdot.rs:21:13
+   |
+21 |             PersonalityInventory { expressivity: exp, ... } => exp
+   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing field `instrumentality`
```

Comment:
I agree with @xfix's comment, it'd be nice if this error wasn't shown. IIRC, this error happens during type check, so you'd have to modify the `ast::StructPatKind` and its equivalent in the `hir` to include a boolean `recovered` and only emit this error if it's true (I'm doing the same for type errors involving tail expressions and recovered if blocks). I'm sure that there're are other errors that could benefit from this signal too, but I wouldn't go looking to fix them on this PR. This change would probably cause many more files to be modified, so we could land this first and have a follow up PR if you prefer.